turbopack/crates/turbo-persistence/src/interning_serde/mod.rs

```diff
@@ -1,40 +1,122 @@
 //! Exposed for usage in `turbo-tasks-backend`
 
-use std::io::Write;
+use std::{
+    io::{Read, Write},
+    sync::LazyLock,
+};
 
+use dashmap::DashMap;
 use indexmap::IndexSet;
 use rustc_hash::FxBuildHasher;
-use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use serde::{de::DeserializeOwned, Serialize};
 use turbo_rcstr::RcStr;
 
-#[derive(Serialize, Deserialize)]
-struct Data(Vec<u8>, IndexSet<RcStr, FxBuildHasher>);
+static GLOBAL_INTERN_MAP: LazyLock<DashMap<RcStr, u32>> = LazyLock::new(DashMap::new);
+static GLOBAL_INTERN_MAP_REVERSE: LazyLock<DashMap<u32, RcStr>> = LazyLock::new(DashMap::new);
```

Comment:
These should be "global" to the database, but not actually global to the program. That's where thread-locals should be useful: The thread local can have the pointer to the current maps.

In practice, we only have one database at a time, but unit tests might have multiple instances (e.g. `cargo test` runs tests in parallel threads in the same process space).

Reply:
Oops, I forgot to adjust this part. Thank you!

Reply:
I removed the in-memory cache for now. I'll add it after verifying that it's working correctly